lib/Driver/Job.cpp

```diff
@@ -131,6 +178,22 @@ StringRef CommandOutput::getAdditionalOutputForType(types::ID Type) const {
   return getOutputForInputAndType(Inputs[0].Primary, Type);
 }
 
+SmallVector<StringRef, 16>
+CommandOutput::getAdditionalOutputsForType(types::ID Type) const {
+  SmallVector<StringRef, 16> V;
+  size_t NonEmpty = 0;
+  if (AdditionalOutputTypes.count(Type) != 0) {
+    for (auto const &I : Inputs) {
+      auto Out = getOutputForInputAndType(I.Primary, Type);
+      if (!Out.empty())
+        ++NonEmpty;
+      V.push_back(Out);
```

Comment:
This isn't good enough; if `Out` is empty it needs to not get added to `V`. But it shouldn't have gotten into the output map in the first place, then.

Reply:
Out being empty here means "no entry in the OFM". It does not mean there's an empty string in the OFM.

Reply:
Sure, but you can't push that back into the vector.

Reply:
As I said in the comment above: that's exactly what many of the call sites expect us to be doing in the case of `getPrimaryOutputFilenames`, and `getAdditionalOutputForType` also signals absence with an empty string. So if we made `getAdditionalOutputsForType` not-do that, it'd be behaving inconsistently. I can do that if you like, I just figured it'd be less-perplexing from an API perspective if the two were consistent, and dealt-with consistently in their callers.